src/quo2/components/avatars/wallet_user_avatar/view.cljs

```diff
@@ -0,0 +1,52 @@
+(ns quo2.components.avatars.wallet-user-avatar.view
+  (:require [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as quo.theme]
+            [react-native.core :as rn]
+            [quo2.components.avatars.wallet-user-avatar.style :as style]
+            utils.string))
+
+(def properties
+  {:size-20 {:size        20
+             :font-size   :label
+             :font-weight :medium}
+   :size-24 {:size        24
+             :font-size   :label
+             :font-weight :semi-bold}
+   :size-32 {:size        32
+             :font-size   :paragraph-2
+             :font-weight :semi-bold}
+   :size-48 {:size        48
+             :font-size   :paragraph-1
+             :font-weight :semi-bold}
+   :size-64 {:size        64
+             :font-size   :paragraph-1
+             :font-weight :medium}
+   :size-80 {:size        80
+             :font-size   :heading-1
+             :font-weight :medium}})
+
+(defn- view-internal
+  "Options:
+
+  :full-name - string (default: nil) - used to generate initials
+  :color  - keyword (default: nil) - color of the avatar
+  :size   - keyword (default: last element of properties object) - size of the
+  avatar"
+  [{:keys [full-name customization-color size theme]
+    :or   {size (last (keys properties))}}]
+  (let [circle-size  (:size (size properties))
+        small?       (= size :size-20)
+        circle-color (style/circle-color customization-color)
+        text-color   (style/text-color customization-color theme)]
+
+    [rn/view
+     {:style (style/container circle-size circle-color)}
+     [text/text
+      {:accessibility-label :wallet-user-avatar
+       :size                (:font-size (size properties))
+       :weight              (:font-weight (size properties))
+       :style               (style/text text-color)}
```

Comment:
tbh i also think inline is fine here since its cheap computationally wise, but i also acknowledge that inline react styles smell a little so after this comment https://github.com/status-im/status-mobile/pull/17328#discussion_r1329976145 from i moved style outside

@mmilad75 @J-Son89 what should we do here?

Reply:
I personally would put it in the style file but move the text color function into the styles file too